File: server-v1/src/db/models/user.model.js
const { Model, DataTypes, Sequelize } = require("sequelize");
const USER_TABLE = "user";

const UserSchema = {
  id: {
    allowNull: false,
    autoIncrement: true,
    primaryKey: true,
    type: DataTypes.INTEGER,
  },
  name: {
    allowNull: false,
    type: DataTypes.STRING,
  },
  email: {
    allowNull: false,
    type: DataTypes.STRING,
    unique: true,
  },
  password: {
    allowNull: false,
    type: DataTypes.STRING,
  },
  isAdmin: {
    allowNull: false,
    type: DataTypes.BOOLEAN,
    defaultValue: false,
  },
  image: { // imagen de perfil
    allowNull: true,
    type: DataTypes.STRING,
  },
  viewedProfile: { // cantidad de veces que se ha visitado el perfil
    allowNull: false,
    type: DataTypes.INTEGER,
    defaultValue: 0,
  },
  createdAt: {
    allowNull: false,
    type: DataTypes.DATE,
    defaultValue: Sequelize.NOW,
  },
};

class User extends Model {
  static associate(models) {
    // Relación 1:N entre User y Post
    this.hasMany(models.Post, {
      foreignKey: "userId",
      as: "posts",
    });
    // Relación 1:N entre User y Comment
    this.hasMany(models.Comment, {
      foreignKey: "userId",
      as: "comments",
    });
    // Relación 1:N entre User y Like
    this.hasMany(models.Like, {
      foreignKey: "userId",
      as: "likes",
    });
  }
  static config(sequelize) {
    return {
      sequelize,
      tableName: USER_TABLE,
      modelName: "User",
      timestamps: false,
    };
  }
}

module.exports = { USER_TABLE, UserSchema, User };

File: server-v2/src/models/user_model.py
from datetime import datetime
import config.db as db

db = db.db

# Relacion Many-to-Many entre User y Role
user_roles = db.Table('user_roles',
  db.Column('user_id', db.Integer, db.ForeignKey('user.id'), primary_key=True),
  db.Column('role_id', db.Integer, db.ForeignKey('role.id'), primary_key=True)
)

class User(db.Model):
  id = db.Column(db.Integer, primary_key=True, autoincrement=True)
  username = db.Column(db.String(80), unique=True, nullable=False)
  email = db.Column(db.String(120), unique=True, nullable=False)
  password = db.Column(db.String(120), nullable=False)
  created_at = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
  roles = db.relationship('Role', secondary=user_roles, lazy='subquery',
                            backref=db.backref('users', lazy=True))

  def __init__(self, username, email, password):
    self.username = username
    self.email = email
    self.password = password

